src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/events/CiagInductionTimelineEventTest.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.events
+
+import org.awaitility.kotlin.await
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.verify
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidPrisonNumber
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.TimelineEventType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.assertThat
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.service.events.aValidCiagInductionCreatedEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.service.events.aValidCiagInductionUpdatedEvent
+
+internal class CiagInductionTimelineEventTest : IntegrationTestBase() {
+
+  @Test
+  fun `should process CIAG induction created event`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val inductionCreatedEvent = aValidCiagInductionCreatedEvent(prisonNumber = prisonNumber)
+
+    // When
+    inboundSnsClient.publish(
+      PublishRequest.builder()
+        .topicArn(inboundTopicArn)
+        .message(objectMapper.writeValueAsString(inductionCreatedEvent))
+        .messageAttributes(mapOf("eventType" to MessageAttributeValue.builder().dataType("String").stringValue(inductionCreatedEvent.eventType()).build()))
```

Comment:
I wonder if the call to `PublishRquest.builder` could be abstracted away into a test data factory/builder or similar? That might alleviate some of my concerns about how strongly typed it is(n't 🤣 ) and would make this test method look a little nicer

Re: the objectmapper and being able to inject it into the test data builder, I'd be tempted just to new one up in the test data builder, rather than inject the one from the context - then you can remove it from the IntegrationTestBase superclass